generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/PayloadResponseFactory.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.services.adapter.rest.processor;
+
+import static javax.ws.rs.core.Response.Status.OK;
+import static javax.ws.rs.core.Response.status;
+import static uk.gov.justice.services.common.http.HeaderConstants.ID;
+
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+
+/**
+ * Create {@link Response} with OK status, with resulting JsonEnvelope payload as the returned
+ * entity.
+ */
+public class PayloadResponseFactory implements ResponseFactory {
```

Comment:
I'm not sure it is very clear the difference between this and the EnvelopeResponseFactory.